src/types/message-types.ts

```diff
@@ -16,6 +16,7 @@ export type GenericMessage = {
  */
 export type MessageOptions = {
   dataStream?: Readable;
+  handler?: GenericMessageSubscriptionHandler;
```

Comment:
Consider renaming to `subscriptionHandler` to disambiguate from "message handlers" we already have.